### PR TITLE
fix: updates for PHP8 compatibility

### PR DESCRIPTION
**Pull request by andycowan**
Hi Robin

I've made some adjustments for PHP8 compatibility, and fixed some deprecations that arise from upgrading phpunit.

Some things, particularly what I've done with versions in the composer.json might not be the way you want to do it, so I suggest you have a look and see what you think.

Also, my changes aren't compatible with 7.1 and 7.2 - I haven't removed them from the version constraint - it depends whether you want to carry on supporting them or not...

A.

### CODE REVIEW

**Comment by bepsvpt** [comment]
PHPUnit 8.5.13 supports PHP 8.0 now, I think it does not require to upgrade PHPUnit to 9.x version.

After testing in local, I think it only needs to change `php` and `phpbench/phpbench` requirements.

```diff
     "require": {
-        "php": "^7.1"
+        "php": "^7.1|^8.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^8.5",
-        "phpbench/phpbench": "^0.17",
+        "phpbench/phpbench": "^1.0.0-alpha3",
         "symfony/phpunit-bridge": "^5.1"
     }
```

File: composer.json [hunk 1]
```diff
@@ -22,11 +22,13 @@
         }
     },
     "require": {
-        "php": "^7.1"
+        "php": ">=7.1"
```
Comment (robinvdvleuten) [inline composer.json:25]: This version is too loosely constrained. Can you make it "^7.2|^8.0" to make sure php 9.0 throws an error.

File: composer.json [hunk 2]
```diff
@@ -22,11 +22,13 @@
...
     },
     "require-dev": {
-        "phpunit/phpunit": "^8.5",
```
Comment (robinvdvleuten) [inline composer.json:28]: PHPUnit 8.5 currently supports both 7.2 and 8.0 so no need to update.

File: composer.json [hunk 3]
```diff
@@ -22,11 +22,13 @@
...
-        "phpbench/phpbench": "^0.17",
+        "phpunit/phpunit": "^9",
+        "phpbench/phpbench": "^1.0.0-alpha2",
         "symfony/phpunit-bridge": "^5.1"
-    }
+    },
+    "minimum-stability": "dev",
+    "prefer-stable": true
```
Comment (robinvdvleuten) [inline composer.json:33]: Please remove both of these lines

File: phpunit.xml.dist [hunk 4]
```diff
@@ -1,23 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (robinvdvleuten) [inline phpunit.xml.dist:1]: Please revert any changes to this file

File: tests/UlidTest.php [hunk 5]
```diff
@@ -24,15 +24,15 @@ public function testGeneratesUppercaseIdentifierByDefault(): void
     {
         $ulid = Ulid::generate();
 
-        $this->assertRegExp('/[0-9][A-Z]/', (string) $ulid);
```
Comment (robinvdvleuten) [inline tests/UlidTest.php:27]: As we still use PHPUnit 8.5, these changes can be reverted

**Review by robinvdvleuten — CHANGES_REQUESTED** [review]

**Comment by robinvdvleuten** [comment]
Thank you @andycowan for this pull request! And thanks @bepsvpt for doing an initial review by running it locally, this saves me a lot of time! 

@andycowan there are some minor changes though. If you could apply them, this PR is good to merge.

**Comment by robinvdvleuten** [comment]
Superseded by #28

**Pull request closed without merging**